rewrite-java-test/src/test/java/org/openrewrite/java/ChangeClassInheritanceTest.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https:www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Tree;
+import org.openrewrite.java.tree.CoordinateBuilder;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Space;
+import org.openrewrite.marker.Markers;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static org.openrewrite.java.Assertions.java;
+
+class ChangeClassInheritanceTest implements RewriteTest {
+
+    @Nested
+    class WithExtendsTest implements RewriteTest {
+        @Override
+        public void defaults(RecipeSpec spec) {
+            spec.recipe(RewriteTest.toRecipe(() -> new JavaIsoVisitor<>() {
+                private final J.Identifier arrayList = new J.Identifier(Tree.randomId(), Space.EMPTY, Markers.EMPTY, ArrayList.class.getName(), JavaType.buildType(ArrayList.class.getName()), null);
+
+                @Override
+                public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration c, ExecutionContext executionContext) {
+                    J.ClassDeclaration classDecl = super.visitClassDeclaration(c, executionContext);
+                    if (classDecl.getExtends() != null && Objects.equals(arrayList.getType(), classDecl.getExtends().getType())) {
+                        return classDecl;
+                    }
+                    return classDecl.withExtends(arrayList.withPrefix(Space.SINGLE_SPACE));
+                }
+            }));
+        }
+
+        @Test
+        void addExtends() {
+            java(
```

Comment:
This test and a few others are missing the wrapping `rewriteRun()`. I think this also explains why you didn't have to change `withExtends()`.

Reply:
Good catch, I will have a look on the two testcases!

Reply:
Fixed it by now and also discovered that I still used the deprecated constructor for identifiers which I changed now as well. Feel free to review again when you are free.